### PR TITLE
[Client Profiles] CHI-2367 Customizable options for blocking timeframes

### PR DESCRIPTION
**Pull request by mythilytm**
Primary reviewer:

## Description
- This PR allows for time based blocks on identifiers
- Set up "Short Term Block (24 Hours)" and "Long Term Block (5 Years)" options. For each option, `flag`, `label`, and `durationInHours` are defined to be used in the frontend
- Custom block durations are sorted and mapped within the UI, allowing administrators to choose and apply them easily.

### Checklist
- [x] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [x] Strings are localized
- [x] Tested for chat contacts
- [x] Tested for call contacts

### Related Issues
Fixes # [2367](https://tech-matters.atlassian.net/browse/CHI-2367), [2368](https://tech-matters.atlassian.net/browse/CHI-2368)

### Verification steps
- Run this branch for backend -  https://github.com/techmatters/hrm/pull/536
- Build `hrm-form-definitions`
- Run Aselo dev locally

- Check options in status flags like below

Screenshot 2024-01-19 at 1 31 46 PM
Screenshot 2024-01-19 at 1 31 55 PM

### CODE REVIEW

File: hrm-form-definitions/src/__tests__/fetchDefinitionsMock.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ const files = [
   'ReferenceData.json',
   'BlockedEmojis.json',
   'profileForms/Sections.json',
+  'profileForms/FlagDurations.json',
```
Comment (GPaoloni) [inline hrm-form-definitions/src/__tests__/fetchDefinitionsMock.ts:51]: Wouldn't make sense to collapse this two into a single JSON file? :thinking:

File: plugin-hrm-form/src/states/profile/hooks/useProfileFlags.ts [hunk 2]
```diff
@@ -107,18 +111,30 @@ export const useProfileFlags = (profileId?: Profile['id']): UseProfileFlags => {
...
+  const combinedProfileFlags = useMemo(() => {
+    return filteredProfileFlags.map(flag => {
+      const matchingFlag = profileFlags.find(pf => pf.id === flag.id);
+
+      return {
+        ...flag,
+        validUntil: flag.validUntil || matchingFlag?.validUntil || null,
+      };
+    });
+  }, [filteredProfileFlags, profileFlags]);
```
Comment (GPaoloni) [inline plugin-hrm-form/src/states/profile/hooks/useProfileFlags.ts:132]: What does `combinedProfileFlags` represents? Is this necessary in the state or is something we can easily compute in the UI when needed?

**Review by GPaoloni — APPROVED** [review]
LGTM! :rocket:

File: plugin-hrm-form/src/states/profile/hooks/useProfileFlags.ts (hunk 2 above)
Comment (mythilytm) [reply plugin-hrm-form/src/states/profile/hooks/useProfileFlags.ts:132]: flag `name` is missing in `profileFlags` which I need for a few places to check the name as included in the form definitions

Comment (mythilytm) [reply plugin-hrm-form/src/states/profile/hooks/useProfileFlags.ts:132]: There are two places where this needed to be calculated. That is why I moved it to this hook

**Pull request merged**